extensions/cloudfiles-extensions/src/test/java/io/druid/storage/cloudfiles/CloudFilesByteSourceTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Druid - a distributed column store.
+ *  Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.druid.storage.cloudfiles;
+
+import org.easymock.EasyMockSupport;
+import org.jclouds.io.Payload;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.easymock.EasyMock.*;
+
+public class CloudFilesByteSourceTest extends EasyMockSupport {
+
+	@Test
+	public void openStreamTest() throws IOException {
+		final String path = "path";
+
+		CloudFilesObjectApiProxy objectApi = createMock(CloudFilesObjectApiProxy.class);
+		CloudFilesObject cloudFilesObject = createMock(CloudFilesObject.class);
+		Payload payload = createMock(Payload.class);
+		InputStream stream = createMock(InputStream.class);
+
+		expect(objectApi.get(path)).andReturn(cloudFilesObject);
+		expect(cloudFilesObject.getPayload()).andReturn(payload);
+		expect(payload.openStream()).andReturn(stream);
+
+		replayAll();
+
+		CloudFilesByteSource byteSource = new CloudFilesByteSource(objectApi, path);
+		byteSource.openStream();
+
+		verifyAll();
+	}
+	
+	@Test(expected = IOException.class)
```

Comment:
For recoverable error can you make it throw only the first call and succeed the second?